src/toast/ops/operator.py

```diff
@@ -134,23 +150,48 @@ def provides(self):
                 or modified.
 
         """
-        return self._provides()
+        # Ensure that all keys exist
+        prov = self._provides()
+        for key in ["metadata", "detdata", "shared", "intervals"]:
+            if key not in prov:
+                prov[key] = list()
+        return prov
 
-    def _accelerators(self):
-        # Do not force descendent classes to implement this.  If it is not
-        # implemented, then it is clear that the class does not support any
-        # accelerators
-        return list()
+    def _supports_acc(self):
+        return False
 
-    def accelerators(self):
-        """List of accelerators supported by this Operator.
+    def supports_acc(self):
+        """Query whether the operator supports OpenACC
 
         Returns:
-            (list):  List of pre-defined accelerator names supported by this
-                operator (and by TOAST).
+            (bool):  True if the operator can use OpenACC, else False.
 
         """
-        return self._accelerators()
+        return self._supports_acc()
+
+    def _use_acc(self, data):
```

Comment:
Are we positive it should not be an error condition to ask for GPU acceleration when not all data is staged to the card?

Reply:
Each operator can support OpenACC, but the decision of whether to use it or not is controlled by the argument to `_exec()`.  This allows us to disable the OpenACC codepath if we know from the calling code that it does not make sense.  Similarly, the Pipeline operator can check if all its operators support accelerators, and can ensure that the inputs are all staged.